src/test/client.rs

```diff
@@ -180,13 +181,72 @@ async fn list_database_names() {
             .unwrap();
     }
 
-    let new_dbs = client.list_database_names(None).await.unwrap();
+    let new_dbs = client.list_database_names(None, None).await.unwrap();
 
     for name in expected_dbs {
         assert_eq!(new_dbs.iter().filter(|db_name| db_name == &name).count(), 1);
     }
 }
 
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn list_authorized_databases() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || !client.auth_enabled() {
+        return;
+    }
+
+    let dbs = &[
+        format!("{}1", function_name!()),
+        format!("{}2", function_name!()),
+    ];
+
+    for name in dbs {
+        client
+            .database(name)
+            .create_collection("coll", None)
+            .await
+            .unwrap();
+        client
+            .create_user(
+                &format!("user_{}", name),
+                "pwd",
+                &[Bson::from(doc! { "role": "readWrite", "db": name })],
+                &[AuthMechanism::ScramSha256],
+            )
+            .await
+            .unwrap();
+    }
+
+    for name in dbs {
+        let mut options = CLIENT_OPTIONS.clone();
+        let credential = Credential::builder()
+            .username(Some(format!("user_{}", name)))
+            .password(Some(String::from("pwd")))
+            .build();
+        options.credential = Some(credential);
+        let client = Client::with_options(options).unwrap();
+
+        let options = ListDatabasesOptions::builder()
+            .authorized_databases(true)
+            .build();
+        let result = client
+            .list_database_names(None, Some(options))
```

Comment:
Similar to above, since the method takes `impl Into<Option<ListDatabaseOptions>>`, you can just put `.list_database_names(None, options)`. We define all of our option parameters this way to make things easier for users.